### PR TITLE
#1086 Added multiline support to signature badge

### LINKED ISSUE #1086: UI - encryption/signature badges to support multiline
> Got this for FlowCrypt/flowcrypt-ios/Core/source/assets/compat/mime-email-plain-signed-detached.txt. Signature error looks too long.

The error is not necessarily too long, but he visual should be fixed so that the badge flows on multiple lines. I'll file a separate issue for that.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/issues/1020#issuecomment-978133605_

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR added multiline support to signature badge

close #1086 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

image
image

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift [hunk 1]
```diff
@@ -36,10 +36,12 @@ struct ProcessedMessage {
             case .unsigned:
                 return "message_not_signed".localized
             case let .error(message):
-                return "message_signature_verify_error".localizeWithArguments(message.lowercasingFirstLetter())
+                let signaturErrorString = "message_signature_verify_error".localized
```
Comment (sosnovsky) [inline FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift:39]: `signaturErrorString` -> `signatureErrorString`

File: FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift [hunk 2]
```diff
@@ -36,10 +36,12 @@ struct ProcessedMessage {
...
+                return signaturErrorString + GeneralConstants.Global.signatureSeparator + message.lowercasingFirstLetter()
             case let .missingPubkey(longid):
-                let message = "message_missing_pubkey".localizeWithArguments(longid)
-                return "message_signature_verify_error".localizeWithArguments(message)
+                let missigngPubKeyString = "message_missing_pubkey".localizeWithArguments(longid.spaced(every: 4))
+                let signaturErrorString = "message_signature_verify_error".localized
```
Comment (sosnovsky) [inline FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift:43]: `missigngPubKeyString` -> `missingPubKeyString`
`signaturErrorString` -> `signatureErrorString`

File: FlowCryptUI/Nodes/BadgeNode.swift [hunk 3]
```diff
@@ -45,18 +53,37 @@ public final class BadgeNode: ASDisplayNode {
...
+        if input.additionalText != nil {
+            showAdditionalText = !showAdditionalText
+            setNeedsLayout()
+        }
     }
 
     override public func layoutSpecThatFits(_: ASSizeRange) -> ASLayoutSpec {
+        let textSpec = ASStackLayoutSpec(
+            direction: .vertical,
+            spacing: 5.0,
```
Comment (sosnovsky) [inline FlowCryptUI/Nodes/BadgeNode.swift:76]: current spacing looks too large, let's try to decrease it to 2 or 3

File: FlowCryptUI/Nodes/BadgeNode.swift [hunk 4]
```diff
@@ -45,18 +53,37 @@ public final class BadgeNode: ASDisplayNode {
         automaticallyManagesSubnodes = true
 
         textNode.attributedText = input.text
+        additionalTextNode.attributedText = input.additionalText
```
Comment (sosnovsky) [inline FlowCryptUI/Nodes/BadgeNode.swift:56]: long error messages are clipped:

Screenshot 2023-09-26 at 11 02 49

File: FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift (hunk 1 above)
Comment (ioanmo226) [reply FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift:39]: Oops. thanks for notice

**Review by sosnovsky — APPROVED** [review]
Looks good 👍

**Pull request merged**